api.go

```diff
@@ -364,40 +366,46 @@ func (api *API) addResource(prototype jsonapi.MarshalIdentifier, source CRUD) *r
 		}
 	}
 
-	api.router.Handle("POST", baseURL, func(w http.ResponseWriter, r *http.Request, params map[string]string) {
-		info := requestInfo(r, api)
-		c := api.contextPool.Get().(APIContexter)
-		c.Reset()
-		api.middlewareChain(c, w, r)
-		err := res.handleCreate(c, w, r, info.prefix, *info)
-		api.contextPool.Put(c)
-		if err != nil {
-			api.handleError(err, w, r)
-		}
-	})
+	if _, ok := source.(CRUDCreate); ok {
```

Comment:
I kinda stumble across the interface names, I know that you just tried to keep it consistent. But maybe CRUD wasn't the best name either. :P 

How about something like:
`ResourceCreator` `ResourceUpdater` or even
`SingleResourceCreator` 

I'm so bad at naming things but imho the interface name should not simply be the name of its method